test/extended/networking/egressip.go

```diff
@@ -0,0 +1,654 @@
+			g.JustBeforeEach(func() {
+				// SCC privileged is needed to run tcpdump on the packet sniffer containers, and at the minimum host networked is needed for
+				// host networked pods.
+				g.By("Adding SCC privileged to the external namespace")
+				_, err := oc.AsAdmin().Run("adm").Args("policy", "add-scc-to-user", "privileged", fmt.Sprintf("system:serviceaccount:%s:default", externalNamespace)).Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("Spawning the packet sniffer pods on the EgressIP assignable hosts")
+				packetSnifferDaemonSet, err = createPacketSnifferDaemonSet(oc, externalNamespace, egressIPNodesOrderedStr, targetProtocol, targetPort)
+				o.Expect(err).NotTo(o.HaveOccurred())
+			})
+
+			g.It("pods should have the assigned EgressIPs and EgressIPs can be deleted and recreated", func() {
+				g.By("Creating the EgressIP test source deployment with number of pods equals number of EgressIP nodes")
+				_, routeName, err := createAgnhostDeploymentAndIngressRoute(oc, egressIPNamespace, "", ingressDomain, len(egressIPNodesOrderedStr), egressIPNodesOrderedStr)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				// For this test, get a single EgressIP per node.
+				// Note: On some clouds like GCP, there is no dedicated CIDR per node and instead all EgressIPs come from a common pool.
+				// Thus, this is only an artificial assignment of EgressIP to node on these cloud platforms and the EgressIP feature
+				// will pick the actual node.
+				g.By("Getting a map of source nodes and potential Egress IPs for these nodes")
+				egressIPsPerNode := 1
+				nodeEgressIPMap, err := findNodeEgressIPs(oc, egressIPNodesOrderedStr, cloudType, egressIPsPerNode)
+				framework.Logf("%v", nodeEgressIPMap)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("Choosing the EgressIPs to be assigned, one per node")
+				var egressIPs []string
+				egressIPSet := make(map[string]struct{})
+				for _, eip := range nodeEgressIPMap {
+					_, ok := egressIPSet[eip[0]]
+					if !ok {
+						egressIPs = append(egressIPs, eip[0])
+						egressIPSet[eip[0]] = struct{}{}
+					}
+				}
+
+				g.By("Marshalling the desired EgressIPs into a string")
+				egressIPsString, err := json.Marshal(egressIPs)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				g.By("Creating the EgressIP object and writing it to disk")
+				egressIPObjectName := egressIPNamespace
+				var egressIPConfig = fmt.Sprintf(
+					egressIPYamlTemplateNamespaceSelector, // template yaml
+					egressIPObjectName,                    // name of EgressIP
+					egressIPsString,                       // compact yaml of egressIPs
+					fmt.Sprintf("kubernetes.io/metadata.name: %s", egressIPNamespace), // namespace selector
+				)
+				err = ioutil.WriteFile(tmpDirEgressIP+"/"+egressIPYaml, []byte(egressIPConfig), 0644)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				// Run this twice to make sure that repeated EgressIP creation and deletion works.
+				iterations := 5
```

Comment:
what is "iterations"? is it num packets sent for the packet sniffer work?

Reply:
I renamed this to: `numberOfRequestsToSend`